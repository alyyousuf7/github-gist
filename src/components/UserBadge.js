import PropTypes from 'prop-types';

import './UserBadge.css';

function UserBadge({ username, avatar }) {
  return <div className="user-badge">
    <img src={avatar} alt={username} /> {username}
  </div>
}

UserBadge.propTypes = {
  username: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
};

export default UserBadge;
