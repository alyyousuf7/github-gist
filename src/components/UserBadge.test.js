
import { render } from '@testing-library/react';

import UserBadge from './UserBadge';

describe('UserBadge', () => {
  it('should render correctly', async () => {
    const { container } = render(<UserBadge username='my username' avatar='http://url-to-avatar' />);

    expect(container).toMatchSnapshot();
  });
});
