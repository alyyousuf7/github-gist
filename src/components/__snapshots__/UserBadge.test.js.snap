// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBadge should render correctly 1`] = `
<div>
  <div
    class="user-badge"
  >
    <img
      alt="my username"
      src="http://url-to-avatar"
    />
     
    my username
  </div>
</div>
`;
